pkg/controller/replication/replication_controller.go

```diff
@@ -105,20 +110,24 @@ type ReplicationManager struct {
 
 	// Controllers that need to be synced
 	queue *workqueue.Type
+
+	// garbageCollectorEnabled denotes if the garbage collector is enabled. RC
+	// manager behaves differently if GC is enabled.
+	garbageCollectorEnabled bool
 }
 
 // NewReplicationManager creates a replication manager
-func NewReplicationManager(podInformer framework.SharedIndexInformer, kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc, burstReplicas int, lookupCacheSize int) *ReplicationManager {
+func NewReplicationManager(podInformer framework.SharedIndexInformer, kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc, burstReplicas int, lookupCacheSize int, garbageCollectorEnabled bool) *ReplicationManager {
 	eventBroadcaster := record.NewBroadcaster()
 	eventBroadcaster.StartLogging(glog.Infof)
 	eventBroadcaster.StartRecordingToSink(&unversionedcore.EventSinkImpl{Interface: kubeClient.Core().Events("")})
-	return newReplicationManagerInternal(
+	return newReplicationManager(
 		eventBroadcaster.NewRecorder(api.EventSource{Component: "replication-controller"}),
-		podInformer, kubeClient, resyncPeriod, burstReplicas, lookupCacheSize)
+		podInformer, kubeClient, resyncPeriod, burstReplicas, lookupCacheSize, garbageCollectorEnabled)
 }
 
-// newReplicationManagerInternal configures a replication manager with the specified event recorder
-func newReplicationManagerInternal(eventRecorder record.EventRecorder, podInformer framework.SharedIndexInformer, kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc, burstReplicas int, lookupCacheSize int) *ReplicationManager {
+// newReplicationManager configures a replication manager with the specified event recorder
```

Comment:
sure, it's an improvement, but it's also noise in a PR this size :) (but
don't change it back)

On Tue, Jul 12, 2016 at 11:08 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/controller/replication/replication_controller.go
> <https://github.com/kubernetes/kubernetes/pull/27600#discussion_r70568476>
> :
>
> >  }
> >
> > -// newReplicationManagerInternal configures a replication manager with the specified event recorder
> > -func newReplicationManagerInternal(eventRecorder record.EventRecorder, podInformer framework.SharedIndexInformer, kubeClient clientset.Interface, resyncPeriod controller.ResyncPeriodFunc, burstReplicas int, lookupCacheSize int) *ReplicationManager {
> > +// newReplicationManager configures a replication manager with the specified event recorder
>
> Function name starting with lower case already suggests its an internal
> function. @bprashanth <https://github.com/bprashanth> suggested this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27600/files/198f8ecb531c17a17ba0bfbfb0b843c7f73cdb38#r70568476>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAnglrgdfOvWmp7zPPTYKd4ZQ9skxIqIks5qVIDDgaJpZM4I4CdG>
> .
>